Core/include/Acts/Definitions/Direction.hpp

```diff
@@ -0,0 +1,161 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2018 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+
+#include <iosfwd>
+#include <string>
+
+namespace Acts {
+
+/// The direction is always with respect to a given momentum, surface normal or
+/// other general axes
+class Direction final {
+ private:
+  enum class Value : int {
+    Negative = -1,
+    Positive = 1,
+  };
+
+ public:
+  static constexpr auto Negative = Value::Negative;
+  static constexpr auto Positive = Value::Positive;
+
+  static constexpr auto Backward = Value::Negative;
+  static constexpr auto Forward = Value::Positive;
+
+  /// This turns a signed value into a direction. Will assert on zero.
+  ///
+  /// @param scalar is the signed value
+  /// @param zeroAsPositive if set, 0 will be handled as "positive"
+  ///
+  /// @return a direciton enum
+  static inline constexpr Direction fromScalar(ActsScalar scalar,
+                                               bool zeroAsPositive = false) {
+    if (!zeroAsPositive) {
+      assert(scalar != 0);
+    }
+    return scalar >= 0 ? Value::Positive : Value::Negative;
+  }
```

Comment:
Would it be a good idea to have a separate function `fromScalarZeroAsPositive` to make it super extra clear, or is the argument enough?